twenty-first/src/math/polynomial.rs

```diff
@@ -1360,6 +1416,403 @@ where
+        let even_interpolant =
+            Self::fast_modular_coset_interpolate(&even_domain_targets, offset, modulus);
+        let odd_interpolant =
+            Self::fast_modular_coset_interpolate(&odd_domain_targets, offset * omega, modulus);
+
+        // 6. Multiply with zerofiers and add.
+        let interpolant = even_interpolant
+            .multiply(&preprocessed.odd_zerofiers[(n / 2).ilog2() as usize])
+            + odd_interpolant.multiply(&preprocessed.even_zerofiers[(n / 2).ilog2() as usize]);
+
+        // 7. Reduce by modulus and return.
+        interpolant.reduce(modulus)
+    }
+
+    /// Extrapolate a Reed-Solomon codeword, defined relative to a coset of the
+    /// subgroup of order n (codeword length), in new points.
+    pub fn coset_extrapolate(
+        domain_offset: BFieldElement,
+        codeword: &[FF],
+        points: &[FF],
+    ) -> Vec<FF> {
+        if points.len() < Self::FAST_COSET_EXTRAPOLATE_THRESHOLD {
+            Self::fast_coset_extrapolate(domain_offset, codeword, points)
+        } else {
+            Self::naive_coset_extrapolate(domain_offset, codeword, points)
+        }
+    }
+
+    fn naive_coset_extrapolate_preprocessing(points: &[FF]) -> (ZerofierTree<FF>, Vec<FF>, usize) {
+        let zerofier_tree = ZerofierTree::new_from_domain(points);
+        let (shift_coefficients, tail_length) =
+            Self::shift_factor_ntt_with_tail_size(&zerofier_tree.zerofier());
+        (zerofier_tree, shift_coefficients, tail_length)
+    }
+
+    fn naive_coset_extrapolate(
+        domain_offset: BFieldElement,
+        codeword: &[FF],
+        points: &[FF],
+    ) -> Vec<FF> {
+        let n = codeword.len();
+        let logn = n.ilog2();
+        let mut coefficients = codeword.to_vec();
+        intt(
+            &mut coefficients,
+            BFieldElement::primitive_root_of_unity(n as u64).unwrap(),
+            logn,
+        );
+        let interpolant =
+            Polynomial::new(coefficients).scale(FF::from(domain_offset.inverse().value()));
+        interpolant.batch_evaluate(points)
+    }
+
+    fn fast_coset_extrapolate(
+        domain_offset: BFieldElement,
+        codeword: &[FF],
+        points: &[FF],
+    ) -> Vec<FF> {
+        let zerofier_tree = ZerofierTree::new_from_domain(points);
+        let minimal_interpolant = Self::fast_modular_coset_interpolate(
+            codeword,
+            domain_offset,
+            &zerofier_tree.zerofier(),
+        );
+        minimal_interpolant.divide_and_conquer_batch_evaluate(&zerofier_tree)
+    }
+
+    /// Extrapolate many Reed-Solomon codewords, defined relative to the same
+    /// coset of the subgroup of order `codeword_length`, in the same set of
+    /// new points.
+    ///
+    /// #Example
+    /// ```
+    /// use twenty_first::prelude::Polynomial;
+    /// use twenty_first::prelude::BFieldElement;
+    /// use twenty_first::prelude::bfe_vec;
+    /// use twenty_first::prelude::bfe;
+    ///
```

Comment:
To focus on the relevant part of the example, consider using `/// #`. This prevents that line from appearing in the rendered document while still being executed in doc tests.

```suggestion
    /// # use twenty_first::prelude::*;
```